### PR TITLE
esptool 2.5.1 (new formula)

### PR DESCRIPTION
**Pull request by jpraczyk**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/esptool.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    url "https://files.pythonhosted.org/packages/cc/74/11b04703ec416717b247d789103277269d567db575d2fd88f25d9767fe3d/pyserial-3.4.tar.gz"
+    sha256 "6e2d401fdee0eab996cf734e67773a0143b932772ca8b42451440cfed942c627"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match(/#{version}/, shell_output("#{bin}/esptool.py version"))
```
Comment (javian) [inline Formula/esptool.rb]: Can we create a test that checks the functionality of the software instead of this ?

Comment (jpraczyk) [reply Formula/esptool.rb]: Unfortunately this software requires specialized hardware to do proper checks (e.g. image flashing, flash erasing) . I've tried coming up with something better than this, but I'm afraid it's the best we've got for now.

Comment (fxcoudert) [reply Formula/esptool.rb]: Try performing an operation and check that it fails.

Comment (javian) [reply Formula/esptool.rb]: Wasn't there some function to transform images that could be used ?

Comment (jpraczyk) [reply Formula/esptool.rb]: @javian Indeed, but those binaries are quite big. Including them in the formula as a base64-encoded string seems pointless. However, there seem to be very small hello world binaries here: https://github.com/espressif/esptool/tree/master/test/images so I took one of them to test image reading functionality of this tool.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @jpraczyk for the pull request!

**Comment by jpraczyk** [comment]
@fxcoudert Thanks a lot for merging it! :)

**Pull request closed without merging**